### ISSUE TITLE
Since v0.11.0, Happy-DOM window/self/globalThis functions crash when not invoked with explicit 'this'

### ISSUE THREAD

**Issue by danielweck** [vitest-dev/vitest#1259]
### Describe the bug

Vitest v0.10.5 worked fine. This commit / PR introduced a bug:
https://github.com/vitest-dev/vitest/pull/1256/files

I am able to reproduce with Preact / Hooks, due to usage of `requestAnimationFrame`, `cancelAnimationFrame` etc. without explicit `this` binding (e.g. `requestAnimationFrame()` instead of `self.requestAnimationFrame()` or `window.requestAnimationFrame()`).

Specifically, the crash occurs here (`this` is `undefined`):
https://github.com/capricorn86/happy-dom/blob/ae900ea746126d04103f9fbaef044bc1a2a15bf5/packages/happy-dom/src/window/Window.ts#L489

My current workaround is to ensure the `this` binding in Vitest's `beforeEach`:

```
window.requestAnimationFrame = window.requestAnimationFrame.bind(window);
window.cancelAnimationFrame = window.cancelAnimationFrame.bind(window);
```

### Reproduction

Simply invoke `requestAnimationFrame()` without explicit `this` binding in a Vitest test => crash.

### System Info

```shell
System:
    OS: macOS 12.3.1
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 3.43 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.5 - ~/.yarn/bin/yarn
    npm: 8.8.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 101.0.4951.54
    Firefox: 100.0
    Safari: 15.4
  npmPackages:
    vitest: ^0.11.0 => 0.11.0 
```
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
I think this PR introduced a bug: https://github.com/vitest-dev/vitest/pull/1258/files

**Comment by sheremet-va** [comment]
> Vitest v0.10.5 worked fine

Previous version also doesn't work for me. I think we always had this issue with happy-dom. Can you confirm?

See: https://stackblitz.com/edit/vitest-dev-vitest-9sqxn4?file=test/suite.test.ts

**Comment by danielweck** [comment]
Thank you for the quick turnaround, @sheremet-va and @antfu :)
I look forward to trying out the next point release.

**Comment by danielweck** [comment]
> > Vitest v0.10.5 worked fine
> 
> Previous version also doesn't work for me. I think we always had this issue with happy-dom. Can you confirm?
> 
> See: https://stackblitz.com/edit/vitest-dev-vitest-9sqxn4?file=test/suite.test.ts

The bug occurred at my end only when I switched from Vitest v0.10.5 to v0.11.0 (I use PNPM, so I was able to quickly run a regression test by re-installing all the dependencies from a fresh lockfile, using the two different versions of Vitest).

Note that unlike your StackBlitz repro, I do not have `@vitest/ui` and `vite` dependencies (just `happy-dom` `3.1.1`). I use Preact with `@testing-library/dom` + `@testing-library/preact` (all on latest versions).
